include/drivers/spi.h

```diff
@@ -129,11 +130,17 @@ extern "C" {
  *    to act as a CS line
  * @param delay is a delay in microseconds to wait before starting the
  *    transmission and before releasing the CS line
+ * @param gpio_dt_flags is the devicetree flags corresponding to how the CS
+ *    line should be driven. GPIO_ACTIVE_LOW and GPIO_ACTIVE_HIGH are ignored
+ *    in favor of SPI_CS_ACTIVE_HIGH and SPI_CS_ACTIVE_LOW options in struct
+ *    spi_config, which also apply to non-GPIO CS controllers.
  */
 struct spi_cs_control {
 	struct device	*gpio_dev;
-	uint32_t		gpio_pin;
 	uint32_t		delay;
+	gpio_pin_t		gpio_pin;
+	gpio_dt_flags_t		gpio_dt_flags;
+	uint16_t		_unused;
```

Comment:
`_unused` really shouldn't be here IMO; it serves only as documentation of a belief that there's unused space at the end of the structure.  If for some reason one of the types changes size, or fields were added above, it would waste space by wrapping into the next word.

Reply:
Can I let you and @tbursztyka resolve that? I added it at his request.

Reply:
I usually prefer to align the structures for readability, but this is a public API and now that indeed we have 2 custom types in it, you can remove it yes.